wgpu-core/src/command/memory_init.rs

```diff
@@ -148,11 +148,15 @@ pub(crate) fn fixup_discarded_surfaces<
 
 impl<A: hal::Api> BakedCommands<A> {
     // inserts all buffer initializations that are going to be needed for executing the commands and updates resource init states accordingly
+    #[allow(unused_variables, unreachable_code)]
     pub(crate) fn initialize_buffer_memory(
         &mut self,
         device_tracker: &mut TrackerSet,
         buffer_guard: &mut Storage<Buffer<A>, id::BufferId>,
     ) -> Result<(), DestroyedBufferError> {
+        #[cfg(metal)]
```

Comment:
Let's implement it properly. wgpu-core doesn't need any platform-specific hacks outside of the backend selection process.
A proper way would be for the wgpu-hal backend to offer a feature "my allocated buffers are always initialized to zero".  And then wgpu-core would check for this feature and activate it.